### PR TITLE
Update parser and markets for Berlin (published 19.02.2020).

### PR DESCRIPTION
**Pull request by johnjohndoe**
- Adapt "properties" parser to format changes in data format as of 19.02.2020.  
  - The "location" property has been replaced with "plz" and "strasse".
  - See `preprocessing/berlin/raw/markets-berlin.json`.
- Update markets for Berlin (published 19.02.2020).

### CODE REVIEW

**Comment by johnjohndoe** [comment]
@mojoaxel Do you have time to check this?

**Comment by johnjohndoe** [comment]
@ubergesundheit Thanks. You are welcome to merge.

File: preprocessing/berlin/compile-berlin-geojson.js [hunk 1]
```diff
@@ -77,10 +77,13 @@ function getWellFormedCoordinates(feature) {
 function prepareFeatureProperties(feature) {
 	var outputProperties = {};
 	var inputProperties = feature.properties.data;
-	var location = inputProperties.location;
-	var title = getDescriptiveTitle(feature.properties.title, location);
+	var zipCode = inputProperties.plz;
+	var streetNameAndNumber = inputProperties.strasse;
+	var sanitizedZipCode = getSanitizeString(zipCode);
+	var sanitizedStreetNameAndNumber = getSanitizeString(streetNameAndNumber);
+	var sanitizedLocation = getLocationText(sanitizedStreetNameAndNumber, sanitizedZipCode);
```
Comment (mojoaxel) [inline preprocessing/berlin/compile-berlin-geojson.js]:
```suggestion
	var sanitizedLocation = getLocationText(streetNameAndNumber, zipCode);
```

File: preprocessing/berlin/compile-berlin-geojson.js [hunk 2]
```diff
@@ -106,6 +109,20 @@ function prepareFeatureProperties(feature) {
...
+function getLocationText(streetNameAndNumber, zipCode) {
+	var location = "";
+	if (streetNameAndNumber !== undefined && streetNameAndNumber !== "") {
+		location = streetNameAndNumber;
+	}
+	if (zipCode !== undefined && zipCode !== "") {
+		location += ", " + zipCode;
+	}
+	return location;
+}
```
Comment (mojoaxel) [inline preprocessing/berlin/compile-berlin-geojson.js]: If `streetNameAndNumber` is undefined but `zipCode` is defined the function retunes something like: ", 90766"

```suggestion
function getLocationText(streetNameAndNumber, zipCode) {
  return [streetNameAndNumber, zipCode].filter(t => t && t.length).join(', ');
}
```

**Comment by mojoaxel** [comment]
too slow...

File: preprocessing/berlin/compile-berlin-geojson.js (hunk 1 above)
Comment (johnjohndoe) [reply preprocessing/berlin/compile-berlin-geojson.js]: Why?

File: preprocessing/berlin/compile-berlin-geojson.js (hunk 2 above)
Comment (johnjohndoe) [reply preprocessing/berlin/compile-berlin-geojson.js]: Thanks. I will change this.

**Comment by johnjohndoe** [comment]
Ups. @mojoaxel PRs are welcome!

File: preprocessing/berlin/compile-berlin-geojson.js (hunk 1 above)
Comment (mojoaxel) [reply preprocessing/berlin/compile-berlin-geojson.js]: My mistake. I confused `getLocationText` with `getSanitizeString`

File: preprocessing/berlin/compile-berlin-geojson.js (hunk 2 above)
Comment (johnjohndoe) [reply preprocessing/berlin/compile-berlin-geojson.js]: @mojoaxel Do you like to prepare a pull request for your suggested change? If so, please re-run the preprocessor with the Berlin data to see if something changes.

Comment (johnjohndoe) [reply preprocessing/berlin/compile-berlin-geojson.js]: @mojoaxel Ping.

Comment (johnjohndoe) [reply preprocessing/berlin/compile-berlin-geojson.js:124]: @mojoaxel Feel free to change this. I won't have time the next weeks myself.

Comment (johnjohndoe) [reply preprocessing/berlin/compile-berlin-geojson.js:124]: :heavy_check_mark: Address here: https://github.com/wo-ist-markt/wo-ist-markt.github.io/pull/335.

**Pull request merged**